src/main/scala/scalaz/nio/package.scala

```diff
@@ -0,0 +1,17 @@
+package scalaz.nio
+
+import java.io.IOException
+
+import scalaz.zio.IO
+
+package object io {
+
+  implicit class IOObjOps(private val ioObj: IO.type) extends AnyVal {
```

Comment:
@jczuchnowski I'm not sure if this has really more advantages over just `IO.syncCatch` in place. But I leave this decision to you!

Reply:
No problem. It does save quite a lot of lines of code, but I did have some doubts myself about it.

Reply:
super cool @jczuchnowski. thank you!

Reply:
Actually I got an idea from @jdegoes workshops yesterday.
In this package object we could have this partial function:
```scala
val JustIOException: PartialFunction[Throwable, IOException] = {
  case io: IOException => io
}
```
that would be used to narrow the error type like:
```scala
IO.syncCatch(channel.finishConnect())(JustIOException)
```
Does it make sense?